### PR TITLE
feat(common): explicitly set the namespace on all the objects

### LINKED ISSUE #441: Explicitly Define `namespace` on all objects (if no `$objectData.namespace` is given use `.Release.Namespace`)
### Is your feature request related to a problem?

See title

### Describe the solution you'd like

See title

### Describe alternatives you've considered

See title

### Additional context

_No response_

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

### PR DESCRIPTION
**Pull request by stavros-k**
**Description**

⚒️ Fixes  #441 

- Added a function to "generate" the namespace and also validate in terms of length and format.

Current blockers:
Some objects added after the initial refactor use a different code style and currently is not straightforward to use the above function.

Those will have to be revisited once they have some tests for themselves.
And after everything else is working and have tests in place (in regards with the namespace change)

TODO:
  - [x] adjusts regex for validation on name and namespace. 
  - [x] Doc updates

Tests:
  - [x] Configmap
  - [x] CronJob
  - [x] DaemonSet
  - [x] Deployment
  - [x] Endpoint
  - [x] EndpointSlice
  - [x] Job
  - [x] NetworkAttachmentDefinition (Due to being only used in Scale and the way the config is fetched, only global and root level namespaces are used. You just can't define the NS on the object itself.)
  - [x] PesistentVolumeClaim
  - [x] RBAC
  - [x] Secret
  - [x] Service
  - [x] ServiceAccount
  - [x] StatefulSet

Objects with the implementation (but no tests):
Also those objects do not have per object override and do not support TPL.
The above issues will be fixed on each one after they are refactored.
  - [x] Certificate
  - [x] Cluster
  - [x] Pooler
  - [x] HorizontalPodAutoscaler
  - [x] Ingress
  - [x] NetworkPolicy
  - [x] PodMonitor
  - [x] PrometheusRule
  - [x] Route
  - [x] ServiceMonitor

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [x] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/lib/metadata/_namespace.tpl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{{- define "tc.v1.common.lib.metadata.namespace" -}}
+  {{- $caller := .caller -}}
+  {{- $objectData := .objectData -}}
+  {{- $rootCtx := .rootCtx -}}
+
+  {{- $namespace := $rootCtx.Release.Namespace -}}
+  {{- if $objectData.namespace -}}
+    {{- $namespace = tpl $objectData.namespace $rootCtx -}}
+  {{- end -}}
```
Comment (Ornias1993) [inline library/common/templates/lib/metadata/_namespace.tpl:9]: can we also take `$rootCtx.Values.global.namespace` and `$rootCtx.Values.namespace` into account?

Comment (Ornias1993) [reply library/common/templates/lib/metadata/_namespace.tpl:9]: background:
`$rootCtx.Values.namespace` can be used if you want a dependency chart to be installed in a different namespace from the main chart

`$rootCtx.Values.global.namespace` can be used if you want all the charts (main and dependency) to use a namespace different than the one helm tries to assign it to (override basically)

whereas `$objectData.namespace` only covers a single object.

All 3 are valid usecases.

File: library/common/templates/lib/metadata/_namespace.tpl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{{- define "tc.v1.common.lib.metadata.namespace" -}}
+  {{- $caller := .caller -}}
+  {{- $objectData := .objectData -}}
+  {{- $rootCtx := .rootCtx -}}
+
+  {{- $namespace := $rootCtx.Release.Namespace -}}
+  {{- with $rootCtx.Values.global.namespace -}}
+    {{- $namespace = tpl . $rootCtx -}}
+  {{- end -}}
```
Comment (Ornias1993) [inline library/common/templates/lib/metadata/_namespace.tpl:9]:
```suggestion
  {{- with $rootCtx.Values.global.namespace -}}
    {{- $namespace = tpl . $rootCtx -}}
  {{- end -}}
  {{- with $rootCtx.Values.namespace -}}
    {{- $namespace = tpl . $rootCtx -}}
  {{- end -}}
```

File: library/common/templates/lib/metadata/_namespace.tpl (hunk 1 above)
Comment (stavros-k) [reply library/common/templates/lib/metadata/_namespace.tpl]: So now its .Release.Namespace -> Global -> Local 
Priority to the one that is most right.

File: library/common/templates/lib/metadata/_namespace.tpl (hunk 2 above)
Comment (stavros-k) [reply library/common/templates/lib/metadata/_namespace.tpl:9]: Fixed.

File: library/common/templates/lib/metadata/_namespace.tpl (hunk 1 above)
Comment (Ornias1993) [reply library/common/templates/lib/metadata/_namespace.tpl]: Yes :)

**Comment by Ornias1993** [comment]
@stavros-k ETA for this?

**Comment by Ornias1993** [comment]
> This weekend

Cool thanks for the info! :)

**Pull request merged**